src/autocropper.py

```diff
@@ -0,0 +1,64 @@
+import cv2
+
+__author__ = 'Thomas'
+
+def iterate_image(img):
+    for i in range(img.shape[0]):
+        for j in range(img.shape[1]):
+            yield (i, j)
+
+class AutoCropper:
+    
+    def __init__(self, blur_size=17, blur_amt=1500, threshold=250, padding=10, pre_crop=10):
+        """
+        Parameters
+        ----------
+        blur_size : int
+            Blur filter width and height.
+        blur_amt : int
+            Gaussian blur strength.
+        threshold : int
+            Only consider white pixels above this threshold for the region of interest.
+        padding : int
+            Expand cropped area by padding on all four sides.
+        pre_crop : int
+            Flat crop around edges before autocropping.
+        """
+        self.blur_size = blur_size
+        self.blur_amt = blur_amt
+        self.threshold = threshold
+        self.padding = padding
+        self.pre_crop = pre_crop
+        self.CROP_FAILURE_THRESHOLD = 10
+    
+    def _prepare_mask(self, mask):
+        # blur out some noise below "white" threshold
+        mask = cv2.GaussianBlur(mask, (self.blur_size, self.blur_size), self.blur_amt)
+        return mask
+    
+    def crop(self, img, mask):
+        # crop flat amound around edge
+        img = img[self.pre_crop:-self.pre_crop, self.pre_crop:-self.pre_crop]
+        mask = mask[self.pre_crop:-self.pre_crop, self.pre_crop:-self.pre_crop]
+        mask = self._prepare_mask(img)
+        
+        bounds_min = [mask.shape[i] for i in [0, 1]]
+        bounds_max = [0, 0]
+        
+        for i, j in iterate_image(mask):
+            # mask is B&W, W is object of interest
+            if mask[i, j] > self.threshold:
+                bounds_min[0] = min(bounds_min[0], i)
+                bounds_max[0] = max(bounds_max[0], i)
+                bounds_min[1] = min(bounds_min[1], j)
+                bounds_max[1] = max(bounds_max[1], j)
+                
+        for i in [0, 1]:
+            bounds_min[i] = max(bounds_min[i] - self.padding, 0)
+            bounds_max[i] = min(bounds_max[i] + self.padding, mask.shape[i] - 1)
```

Comment:
I try to add a quick comment to every loop or conditional statement that's longer than one line, or which has a purpose that isn't immediately obvious without having a full understanding of the surrounding code.
E.g. 
```
# resize boundaries of mask based on light/dark pixels
for i, j in iterate_image(mask):
    ...
```
```
# I don't actually understand what this loop does
for i in [0,1]:
    ....
```